src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -948,7 +968,7 @@ private function addLockSection(ArrayNodeDefinition $rootNode)
                             ->end()
                             ->beforeNormalization()
                                 ->ifTrue(function ($v) { return \is_array($v) && array_keys($v) === range(0, \count($v) - 1); })
-                                ->then(function ($v) { return ['default' => $v]; })
+                                ->then(function ($v) { return ['default' => 1 === \count($v) && isset($v[0]) && \is_array($v[0]) ? $v[0] : $v]; })
```

Comment:
what is the use case that this solves?

Reply:
@jderusse This covers the following test case:

```php
yield [null, ['enabled' => true, 'resources' => ['default' => [class_exists(SemaphoreStore::class) && SemaphoreStore::isSupported() ? 'semaphore' : 'flock']]]];
```

In other words if someone did something like:

```php
    $container->loadFromExtension('framework', [
        'lock' => [
            'enabled' => true,
            ['redis://r1.docker', 'redis://r2.docker'],
        ],
    ]);
```

Reply:
@jderusse Copied the wrong test case, fixed it now.